Modules/Loadable/Markups/MRML/vtkMRMLMarkupsCurveNode.cxx

```diff
@@ -1410,6 +1437,59 @@ void vtkMRMLMarkupsCurveNode::OnCurvatureMeasurementModified(
   self->CurveMeasurementsCalculator->Update();
 }
 
+//---------------------------------------------------------------------------
+void vtkMRMLMarkupsCurveNode::OnTorsionMeasurementModified(
+  vtkObject* caller, unsigned long vtkNotUsed(eid), void* clientData, void* vtkNotUsed(callData))
+{
+  vtkMRMLMarkupsCurveNode* self = reinterpret_cast<vtkMRMLMarkupsCurveNode*>(clientData);
+  vtkMRMLStaticMeasurement* measurement = reinterpret_cast<vtkMRMLStaticMeasurement*>(caller);
+  if (!self || !measurement)
+    {
+    return;
+    }
+
+  if (!measurement->GetEnabled())
+    {
+    // measurement is disabled
+    measurement->ClearValue();
+    if (!self->CurveMeasurementsCalculator->GetCalculateTorsion())
+      {
+      // no need to compute and it was not computed, nothing to do
+      return;
+      }
+    // Disable curve measurement calculator if no torsion metric is needed anymore
+    bool isTorsionComputationNeeded = false;
+    for (int index = 0; index < self->Measurements->GetNumberOfItems(); ++index)
+      {
+      vtkMRMLMeasurement* currentMeasurement = vtkMRMLMeasurement::SafeDownCast(self->Measurements->GetItemAsObject(index));
+      if (currentMeasurement->GetEnabled()
+        && (currentMeasurement->GetName() == self->CurveMeasurementsCalculator->GetAverageTorsionName()
+          || currentMeasurement->GetName() == self->CurveMeasurementsCalculator->GetMaxTorsionName()))
+        {
+        isTorsionComputationNeeded = true;
+        break;
+        }
+      }
+    if (!isTorsionComputationNeeded)
+      {
+      self->CurveMeasurementsCalculator->SetCalculateTorsion(false);
+      self->CurveMeasurementsCalculator->Update();
+      }
+    return;
+    }
+
+  // measurement is enabled
+  if (self->CurveMeasurementsCalculator->GetCalculateTorsion() && measurement->GetValueDefined())
+    {
+    // measurement was already on, nothing to do
+    return;
```

Comment:
How do we know that recomputation is not needed? Shouldn't we check some MTime values?

Reply:
This function handles only enabling/disabling the specific measurements. We discussed it with @lassoan and this part is OK (except that there was a missing call so good that this part has been investigated)